core/node/provider.go

```diff
@@ -115,3 +118,9 @@ func SimpleProviders(reprovideStrategy string, reprovideInterval string) fx.Opti
 		fx.Provide(SimpleReprovider(reproviderInterval)),
 	)
 }
+
+func pinnedProviderStrategy(onlyRoots bool) interface{} {
+	return func(pinner pin.Pinner, dag ipld.DAGService) simple.KeyChanFunc {
```

Comment:
@magik6k @Kubuxu Any thoughts on a preferred alternative to this?

The reason for this is that the "old" reprovider depends on `pin.Pinner` which is a module within `go-ipfs`. To break that dependency I introduced a `Pinner` type in the `go-ipfs-provider` repository, but that is not the type of `Pinner` that's in the dependency injection container. To get around this I do the constructing of the dependency here, within `go-ipfs`, so that I can depend on `pin.Pinner`. The benefit, as I see it, is that this change is relatively small.

We could move the `Pinner` out as well into its own repository, but I really don't want to take on that large of a change right now.

Reply:
Would it be possible to leave the `NewPInnedProvider` thing as a provider implementation within go-ipfs and extract the more generalistic ones (namely `BlockstoreProvider`). This looks very weird right now, and I don't need the PinnedProvider anyways.

Reply:
Offline @magik6k said he's ok with this. Ultimately, he has some work in progress in https://github.com/ipfs/go-ipfs/pull/6387 that will change how all of this works for the better.

Reply:
@hsanjuan I think that makes sense. Let me try it. And now that I think about it, `ipfs-lite` maybe doesn't have the concept of a `Pinner`, is that true?

Reply:
Yeah that's true, no Pinner. Pinner is a go-ipfs specific concept (for now)